### PR TITLE
Load more data movie

### PR DESCRIPTION
**Pull request by kienngo**
Load more data movie
https://edu-redmine.framgia.vn/issues/551

### CODE REVIEW

File: app/src/main/java/com/example/admin/moviebd/screen/BaseRecyclerViewAdapter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public void addData(List<Movie> dataViews) {
+        mListData.addAll((Collection<? extends T>) dataViews);
+        notifyDataSetChanged();
+    }
+
+    public void addLoadingView() {
+        mListData.add(null);
+        notifyItemInserted(mListData.size() - 1);
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/moviebd/screen/BaseRecyclerViewAdapter.java]: 1 là hardcode đó em. khai báo 1 biến private static final int

File: app/src/main/java/com/example/admin/moviebd/screen/BaseRecyclerViewAdapter.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public void removeLoadingView() {
+        mListData.remove(mListData.size() - 1);
+        notifyItemRemoved(mListData.size());
+    }
+
+    @Override
+    public int getItemCount() {
+        return mListData.size();
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/moviebd/screen/BaseRecyclerViewAdapter.java]: check null

File: app/src/main/java/com/example/admin/moviebd/screen/RecyclerViewScrollListener.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            mIsLoading = true;
+        }
+    }
+
+    public void resetStateLoading() {
+        this.mIsLoading = false;
+    }
+
+    public abstract void onLoadMore(int page);
+}
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/moviebd/screen/RecyclerViewScrollListener.java]: need an endline

**Comment by kienngo** [comment]
ready

**Pull request merged**